frontend/src/app/shared/dialog/objective-dialog/objective-form.component.ts

```diff
@@ -93,23 +98,22 @@ export class ObjectiveFormComponent implements OnInit, OnDestroy {
       state: state
     } as unknown as Objective;
 
-    const submitFunction = this.getSubmitFunction(this.data.objective.objectiveId!, objectiveDTO);
+    const submitFunction = this.getSubmitFunction(this.data.objective.objectiveId, objectiveDTO);
     submitFunction.subscribe((savedObjective: Objective) => {
-      this.closeDialog(savedObjective, false, value.createKeyResults!);
+      this.closeDialog(savedObjective, false, value.createKeyResults ?? undefined);
```

Comment:
undefined as fallback?
is this really intended?

Reply:
undefined turns into false because of logic in closeDialog()